pkgs/development/haskell-modules/generic-builder.nix

```diff
@@ -177,7 +197,15 @@ assert allPkgconfigDepends != [] -> pkgconfig != null;
 stdenv.mkDerivation ({
   name = "${pname}-${version}";
 
-  outputs = if (args ? outputs) then args.outputs else ([ "out" ] ++ (optional enableSeparateDataOutput "data") ++ (optional enableSeparateDocOutput "doc"));
+  outputs = if (args ? outputs) then args.outputs else
+    ( [ "out" ]
+    ++ (optional enableSeparateDataOutput "data")
+    ++ (optional enableSeparateDocOutput "doc")
+    ++ (optional enableSeparateBinOutput "bin")
+    ++ (optional enableSeparateBinOutput "libexec")
```

Comment:
That's a lot of outputs. Does e.g. the `libexec` one actually save any disk space in practice? E.g. if you have a case:

````
/nix/store/...-foo-bin/bin/foo
````
that refers to:
````
/nix/store/...-foo-libexec/libexec/foo-helper
````
then splitting the `libexec` output only helps in the case someone needs only `libexec` but not `bin` (which sounds quite rare to me).

Reply:
Why not propagate inputs from .dev instead?